docs/core/extensions/channels.md

```diff
@@ -0,0 +1,174 @@
+
+#### Full mode behavior
+
+When using a bounded channel, you can specify the behavior the channel will adhere to when the configured bound is reached. The following table lists the full mode behaviors for each <xref:System.Threading.Channels.BoundedChannelFullMode> value:
+
+| Value | Behavior |
+|--|--|
+| <xref:System.Threading.Channels.BoundedChannelFullMode.Wait?displayProperty=nameWithType> | This is the default value. Waits for space to be available in order to complete the write operation. |
+| <xref:System.Threading.Channels.BoundedChannelFullMode.DropNewest?displayProperty=nameWithType> | Removes and ignores the newest item in the channel in order to make room for the item being written. |
+| <xref:System.Threading.Channels.BoundedChannelFullMode.DropOldest?displayProperty=nameWithType> | Removes and ignores the oldest item in the channel in order to make room for the item being written. |
+| <xref:System.Threading.Channels.BoundedChannelFullMode.DropWrite?displayProperty=nameWithType> | Drops the item being written. |
+
+> [!IMPORTANT]
+> Whenever a <xref:System.Threading.Channels.Channel%602.Writer%2A?displayProperty=nameWithType> produces faster than a <xref:System.Threading.Channels.Channel%602.Reader%2A?displayProperty=nameWithType> can consume, the channel's reader may experience _back pressure_.
+
+## Producer APIs
+
+The producer functionality is exposed on the <xref:System.Threading.Channels.Channel%602.Writer%2A?displayProperty=nameWithType>. The producer APIs and expected behavior are detailed in the following table:
+
+| API | Expected behavior |
+|--|--|
+| <xref:System.Threading.Channels.ChannelWriter%601.Complete%2A?displayProperty=nameWithType> | Marks the channel as being complete, meaning no more items will be written to it. |
+| <xref:System.Threading.Channels.ChannelWriter%601.TryComplete%2A?displayProperty=nameWithType> | Attempts to mark the channel as being completed, meaning no more data will be written to it. |
+| <xref:System.Threading.Channels.ChannelWriter%601.TryWrite%2A?displayProperty=nameWithType> | Attempts to write the specified item to the channel. When used with an unbounded channel, this always returns `true` unless the channel's writer signals completion with either <xref:System.Threading.Channels.ChannelWriter%601.Complete%2A?displayProperty=nameWithType>, or <xref:System.Threading.Channels.ChannelWriter%601.TryComplete%2A?displayProperty=nameWithType>. |
+| <xref:System.Threading.Channels.ChannelWriter%601.WaitToWriteAsync%2A?displayProperty=nameWithType> | Returns a <xref:System.Threading.Tasks.ValueTask%601> that will complete when space is available to write an item. |
+| <xref:System.Threading.Channels.ChannelWriter%601.WriteAsync%2A?displayProperty=nameWithType> | Asynchronously writes an item to the channel. When used with an unbounded channel, this will write synchronously. |
```

Comment:
Do we need to be so specific about it writing synchronously?  It's accurate but it's also an implementation detail.  We should be able to change this in the future and have it not break anyone adhering to the docs.